socialhome/content/models.py

```diff
@@ -114,10 +115,17 @@ def save(self, author=None, *args, **kwargs):
         if self.pk:
             # Old instance, bust the cache
             self.bust_cache()
-        elif author:
-            # New with author, set a GUID and author
-            self.guid = uuid4()
-            self.author = author
+        else:
+            if author:
+                # New with author, set a GUID and author
+                self.guid = uuid4()
+                self.author = author
+
+            if self.pinned:
+                max_order = Content.objects.aggregate(Max("order"))["order__max"]
```

Comment:
OK this is closer, but it still needs to get content only for *user who created this current content*, not all content in the database. You need to add a filter before the aggregate to limit to current authoring user.